batchflow/research/research.py

```diff
@@ -229,16 +241,27 @@ def _chunks(self, array, size):
         for i in range(0, len(array), size):
             yield array[i:i + size]
 
-    def run(self, n_reps=1, n_iters=None, workers=1, branches=1, name=None,
-            progress_bar=False, gpu=None, worker_class=None, timeout=5, trails=2):
+    def _cv_split(self, n_splits, shuffle):
+        has_dataset = False
+        for unit in self.executables:
+            if getattr(self.executables[unit], 'dataset', None):
+                has_dataset = True
+                self.executables[unit].dataset.cv_split(n_splits=n_splits, shuffle=shuffle)
+        if not has_dataset:
+            raise ValueError('At least one pipeline must have dataset to perform cross-validation')
+
+    def run(self, n_reps=1, n_iters=None, workers=1, branches=1, n_splits=None, shuffle=False, name=None,
```

Comment:
```suggestion
    def run(self, n_reps=1, n_iters=None, workers=1, branches=1, cv=None, shuffle=False, name=None,
```
??

Reply:
I prefer to name it `n_splits` here because it is exactly a parameter which is used in `cv_split` so it is reasonable to have the same name. Moreover, we use `cv` for folds after splitting.